### PR TITLE
Add a MapType test for checksum function

### PR DESCRIPTION
**Pull request by v-jizhang**
Add a MapType test for checksum function.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by ericyuliu** [comment]
LGTM.

**Review by ericyuliu — APPROVED** [review]
@v-jizhang 
testMap looks good, but when I compare with other tests, they are building the block with multiple items, such as  arrayType 
`createArrayBigintBlock(asList(null, asList(1L, 2L), asList(3L, 4L), asList(5L, 6L)));`

shall we extend that or add another test to cover that ( block with multiple maps) ?

**Comment by ericyuliu** [comment]
Also, lets fix the bug in createMapBlock, where BIGINT is hardcoded, with dynamic type

**Comment by ericyuliu** [comment]
LGTM, thanks for fixing the bug and adding more unit tests

File: presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestChecksumAggregation.java [hunk 1]
```diff
@@ -111,6 +121,29 @@ public void testArray()
...
+        long[] keys = {25895L, 111936L, 162115L, 304417L, 3069541L, 3159396L, 3500641L, 5097671L, 5197414L, 5404804L};
+        LongArrayBlock keyBlock = new LongArrayBlock(10, Optional.empty(), keys);
+        BlockBuilder builder = DOUBLE.createBlockBuilder(null, 10);
+        double[] values = {149912.63, 264526.94, 90483.15, 150554.2, 195291.64, 328452.73, 10938.72, 12740.73, 31276.6, 119651.44};
+        for (int i = 0; i < values.length; i++) {
+            DOUBLE.writeDouble(builder, values[i]);
+        }
+        Block valueBlock = builder.build();
+        int[] offsets = {0, 2, 6, 10};
+        Block mutiMapBlock = fromKeyValueBlock(3, Optional.ofNullable(null), offsets, keyBlock, valueBlock);
```
Comment (rschlussel) [inline presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestChecksumAggregation.java:143]: typo: should say multiMapBlock (missing 'l' from "multi")

**Review by rschlussel — APPROVED** [review]
Looks good.  All the tests in this class are a little funny as the expectedChecksum() function basically duplicates the logic from ChecksumAggregationFunction. But I guess they are valuable in that they ensure that the function generation for different types happens correctly.

**Review by rschlussel — CHANGES_REQUESTED** [review]
requesting changes so i don't accidentally merge before the commits are squashed.

**Review by rschlussel — APPROVED** [review]

**Comment by v-jizhang** [comment]
@prestobot kick off tests

**Comment by v-jizhang** [comment]
@prestobot kick off tests.

**Comment by v-jizhang** [comment]
@rschlussel Could this be merged? Thanks

**Pull request merged**